_posts/2020-07-01-#18991.md

```diff
@@ -7,16 +7,74 @@ authors: [naumenkogs]
+  the latter. There are 2 relevant messages: GETADDR (a sender asks to provide
+  him with network addresses the receiver knows along with some per-address
+  metadata) and ADDR (a receiver responds with this message containing the
+  requested data they want to share with the sender).
+
+- This GETADDR/ADDR protocol is currently used every time a node makes a new
+  connection, after receiving the version message.
+
+- Alternatively, network addresses of the nodes are also propagated in the
+  network in an unsolicited way: from time to time, every node makes a
+  self-announcement which gets propagated across the network. DNS seeds is
+  another way of relaying addresses across the network. Neither of these are very
+  relevant to this PR.
+
+- Every time a node hears about another node in the network, it stores/updates
+  the record in the Address Manager (or AddrMan). This is where all the data
+  about nodes in the network is stored, and this is where a GETADDR receiver
+  looks to construct an ADDR response to the requestor.
+
+- It turns out that one can easily scrape full AddrMan content of any reachable
+  node: a spy just have to connect to a victim and execute GETADDR a bunch of
+  times. This scraped data can then be used to infer something that was supposed
+  to be private about a victim.
+
+- For example, a spy can monitor the victim's AddrMan content in real time and
+  figure out which peers a node is connected to. A spy can also compare the
+  AddrMan content of two virtually different nodes (one identified by Tor address
+  and one identified by IPv4) and figure out it's actually the same physical
+  node.
+
+- This PR is a big first step to fix these privacy issues. If we limit (cache)
+  the leaked portion of AddrMan, these inference activities will become much
+  harder. Caching in this context means that ADDR response (which is only a small
+  subset of a node's AddrMan content) remains same for every GETADDR call during
+  (rougly) a day.
+
 ## Questions
 
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+   NACK?](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)
+   (You're always encouraged to put your PR review on GitHub, even after it has been merged)
+
+2. Do you understand the importance of ADDR relay and specifically GETADDR/ADDR
+   protocol? Which goals this protocol has? Which properties of ADDR relay are
+   important?
+
+3. Do you understand the examples of the attacks described above? Can you think
+   of any more ways to exploit scraping AddrMan? If you find something really
+   dangerous, consider discussing this with some other developer privately first.
+
+4. Do you think the attacks are severe? What are the potential consequences of
+   making the topology (local or global) public?
+
+5. Do you think the suggested PR sufficiently addresses the problem?
+
+6. What are the side-effects of the suggested solution?
+
 -->
```

Comment:
This line needs to be removed (`--->`), it's messing up the html to display the meeting log markup.